h2/src/main/org/h2/mvstore/Page.java

```diff
@@ -125,13 +126,15 @@ public static Page create(MVMap<?, ?> map, long version,
         p.values = values;
         p.children = children;
         p.totalCount = totalCount;
-        if (memory == 0) {
+        MVStore store = map.store;
+        if(store.getFileStore() == null) {
+            p.memory = IN_MEMORY;
```

Comment:
please add comment to memory field doc in this class, that memory is only valid for on-disk storage